modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -165,6 +165,31 @@ public static int currentTime(DataContext ctx) {
         return (int) TypeUtils.toInternal((ExecutionContext<?>) ctx, LocalTime.now(), LocalTime.class);
     }
 
+    /** LEAST2. */
+    public static @Nullable Object least2(Object arg0, Object arg1) {
+        return leastOrGreatest(true, arg0, arg1);
+    }
+
+    /** GREATEST2. */
+    public static @Nullable Object greatest2(Object arg0, Object arg1) {
+        return leastOrGreatest(false, arg0, arg1);
+    }
+
+    private static @Nullable Object leastOrGreatest(boolean least, Object arg0, Object arg1) {
+        if (arg0 == null || arg1 == null) {
+            return null;
```

Comment:
Why least2 and greatest2 both return 'null' if one of arg is 'null' ?

Reply:
This may depends on NullCollation.
See Commons class:
`FRAMEWORK_CONFIG  = Frameworks.newConfigBuilder().....withDefaultNullCollation(NullCollation.HIGH))`

Reply:
I suppose cause it defines bounds (upperBound = leastOrGreatest(..), lowerBound = leastOrGreatest(...)) and in case of one arg is null, bound becomes **null** too.

Reply:
Got it. 'null' means unbounded.